src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -1,151 +1,204 @@
+import { v4 as uuidv4 } from "uuid";
+
 import Button from "../../common/Button/Button";
-import { formatDuration } from "../../common/Tools/formatDuration.jsx";
-import "./CreateCourse.css";
-import {
-  CreateNewCourse,
-  AddAuthorToCourse,
-  CreateAuthor,
-  RemoveAuthorFromSelectList
-} from "./helpers/author.helpers.js";
+import Input from "../../common/Input/Input";
+import { mockedAuthorsList, mockedCoursesList } from "../../constants";
+import formatDuration from "../../helpers/formatDuration";
+
+import styles from "./CreateCourse.module.css";
+import Author from "./components/Author";
 
 import {
   BUTTON_TEXT,
-  LABEL_TEXT,
-  MESSAGES,
   PLACEHOLDER_TEXT,
-} from "../../helpers/constants.js";
-import { useCourseAuthorContext } from "../../helpers/CourseAuthorStore.jsx";
+  LABEL_TEXT,
+  getAlertText,
+  PARAMETERS,
+} from "./createCourseStrings";
+
+function CreateCourse() {
+  const navigate = useNavigate();
 
-export const CreateCourse = ({ onCourseCreated }) => {
-  const { addCourse, authors, addAuthor, setAuthors } =
-    useCourseAuthorContext();
   const [title, setTitle] = useState("");
   const [description, setDescription] = useState("");
+  const [newAuthor, setNewAuthor] = useState("");
   const [duration, setDuration] = useState("");
-  const [courseAuthors, setCourseAuthors] = useState([]);
-  const [newAuthorName, setNewAuthorName] = useState("");
+  const [authorsList, setAuthorsList] = useState(mockedAuthorsList);
+  const [selectedAuthorsList, setSelectedAuthorsList] = useState([]);
 
-  const handleAddAuthorToCourse = (authorId) => {
-    AddAuthorToCourse(authorId, authors, courseAuthors, setCourseAuthors);
-  };
+  const ALERT_TEXT = getAlertText();
 
-  const handleRemoveAuthorFromSelectList = (authorId) => {
-    RemoveAuthorFromSelectList(
-      authorId,
-      courseAuthors,
-      setCourseAuthors,
-      authors,
-      setAuthors,
+  function addCourseAuthor(author) {
+    setSelectedAuthorsList([...selectedAuthorsList, author]);
+    setAuthorsList((current) =>
+      current.filter((item) => item.name !== author.name),
     );
-  };
-
-  const handleCreateAuthor = () => {
-    CreateAuthor(newAuthorName, authors, addAuthor, setNewAuthorName);
-  };
+  }
 
-  const handleCreateCourse = () => {
-    CreateNewCourse(
-      title,
-      description,
-      duration,
-      courseAuthors,
-      addCourse,
-      onCourseCreated,
+  function deleteCourseAuthor(author) {
+    setAuthorsList([...authorsList, author]);
+    setSelectedAuthorsList((current) =>
+      current.filter((item) => item.name !== author.name),
     );
-  };
+  }
+
+  function createNewAuthor(author) {
+    if (author.length < PARAMETERS.AUTHOR_MIN_LENGTH) {
+      alert(ALERT_TEXT.AUTHOR_NAME_TOO_SHORT);
+      return;
+    }
+    const newAuthor = {
+      id: uuidv4(),
+      name: author,
+    };
+    setAuthorsList([...authorsList, newAuthor]);
+    mockedAuthorsList.push(newAuthor);
+  }
+
+  function isValid() {
```

Comment:
Please provide a more meaningful name for such validation function (something like `isValidCreateCoursePayload`)

Reply:
changed